src/modules/transactions/reducers/transactions-data.js

```diff
@@ -24,6 +25,12 @@ export default function (transactionsData = DEFAULT_STATE, action) {
         return p
       }, {})
     case CLEAR_TRANSACTION_DATA:
+      return Object.keys(transactionsData).reduce((p, transactionID) => {
+        if (transactionsData[transactionID].status === PENDING) {
+          p[transactionID] = transactionsData[transactionID]
+        }
+        return p
+      }, {})
     case RESET_STATE:
```

Comment:
clear transaction data is called when browser is redirected to transactions or user goes to transactions, need to retain transactions in pending state.